core/src/main/scala/org/apache/spark/shuffle/hash/BlockStoreShuffleFetcher.scala

```diff
@@ -17,23 +17,22 @@
 
 package org.apache.spark.shuffle.hash
 
-import scala.collection.mutable.ArrayBuffer
-import scala.collection.mutable.HashMap
+import java.io.InputStream
+
+import scala.collection.mutable.{ArrayBuffer, HashMap}
 import scala.util.{Failure, Success, Try}
 
 import org.apache.spark._
-import org.apache.spark.serializer.Serializer
 import org.apache.spark.shuffle.FetchFailedException
 import org.apache.spark.storage.{BlockId, BlockManagerId, ShuffleBlockFetcherIterator, ShuffleBlockId}
 import org.apache.spark.util.CompletionIterator
 
-private[hash] object BlockStoreShuffleFetcher extends Logging {
-  def fetch[T](
+private[shuffle] object BlockStoreShuffleFetcher extends Logging {
```

Comment:
There is currently only a single implementation of `ShuffleReader`, `HashShuffleReader`, so having `BlockStoreShuffleFetcher` as `private[hash]` worked fine. However, if we want to support any new `ShuffleReader` implementations, they are going to need to call the `BlockStoreShuffleFetcher` too so I opened up visibility one level to the `shuffle` namespace.